### PR TITLE
[VideoTranslation] Initial add TypedSpec and swagger

### LINKED ISSUE #29078: [Cognitive Services - Azure AI Services] API Review
New API Review meeting has been requested.

**Service Name**: Cognitive Services - Azure AI Services
**Review Created By**: Dapeng Li
**Review Date**: 05/30/2024 04:00 PM PT
**Release Plan**: [](https://web.powerapps.com/apps/?release-plan-id=)
**PR**: 
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: Video dubbing API to translate video from source locale to target locale.

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=ee056bc3-6a0f-ef11-9f89-00224806999b)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by DapengLi2016**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/cognitiveservices/VideoTranslation/.gitignore
Comment (mikeharder) [inline specification/cognitiveservices/VideoTranslation/.gitignore:1]: You shouldn't need this

**Comment by mikeharder** [comment]
All the errors in check `TypeSpec Validation` are actually caused by the changes under folder `VideoTranslation`.  Start by fixing this error:

```
tspconfig.yaml must define the following property:

linter:
  extends:
    - "@azure-tools/typespec-azure-rulesets/data-plane"
```

https://dev.azure.com/azure-sdk/public/_build/results?buildId=3904744&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=0f4aa080-c997-58ca-481c-4930865ae0f8&l=1219

After this, I also think you have some formatting errors in your tsp files.

**Review by mikeharder — DISMISSED** [review]
Remove .gitignore

File: specification/cognitiveservices/VideoTranslation/main.tsp [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+model AADToken
+  is OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit;
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/authorize";
+      scopes: ["https://cognitiveservices.azure.com/.default"];
+    }
+  ]>;
+
+// alias ops = Azure.Core.StandardResourceOperations;
```
Comment (allenjzhang) [inline specification/cognitiveservices/VideoTranslation/main.tsp:52]: nit: remove

**Comment by mikekistler** [comment]
This API version was reviewed and approved in private repo PR: https://github.com/Azure/azure-rest-api-specs-pr/pull/18399

**Review by mikekistler — APPROVED** [review]
Looks good. 👍

File: specification/cognitiveservices/Speech.VideoTranslation/main.tsp [hunk 2]
```diff
@@ -118,6 +139,15 @@ model TranslationInput {
 
   @doc("Translation video file url.")
   videoFileUrl: url;
+
+  @doc("Speaker count of the video.")
+  speakerCount?: int32;
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/Speech.VideoTranslation/main.tsp:144]: nit, what exactly this is? "number of speakers in the video"?

File: specification/cognitiveservices/Speech.VideoTranslation/main.tsp [hunk 3]
```diff
@@ -118,6 +139,15 @@ model TranslationInput {
...
+
+  @doc("Subtitle max char count per segment.")
+  subtitleMaxCharCountPerSegment?: int32;
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/Speech.VideoTranslation/main.tsp:147]: What is the Char? unicode16? unicode8? or other encoding?

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**